src/iotEdgeExplorer.ts

```diff
@@ -205,7 +205,7 @@ export class IoTEdgeExplorer extends BaseExplorer {
         this._outputChannel.show();
         this.outputLine(label, `Start deployment to [${deviceId}]`);
 
-        const url = `/devices/${deviceId}/applyConfigurationContent?api-version=${Constants.IoTHubApiVersion}`;
+        const url = `/devices/${encodeURIComponent(deviceId)}/applyConfigurationContent?api-version=${Constants.IoTHubApiVersion}`;

```

Comment:
do we need to encodeUri for the whole uri string?

Reply:
No needed, only `deviceId` would have special char like `#` which needs URI encode

Reply:
will # be escaped?

Reply:
# will be encoded.